x-pack/plugins/observability_solution/investigate_app/public/pages/details/contexts/investigation_context.tsx

```diff
@@ -81,6 +85,9 @@ export function InvestigationProvider({
     id: initialInvestigation.id,
     initialInvestigation,
   });
+  const alertOriginInvestigation = alertOriginSchema.safeParse(investigation?.origin);
+  const alertId = alertOriginInvestigation.success ? alertOriginInvestigation.data.id : undefined;
+  const { data: alert } = useFetchAlert({ id: alertId });
```

Comment:
🍰  nit: this logic is required every time we use useFetchAlert(), maybe we can refactor the hook to encapsulate this logic: the hook itself could use useInvestigation to retrieve the investigation, and we won't need to expose the originated alert in this context. I'm already worried about this context becoming bloated with too many things.

`const { data: alert } = useFetchAlertOrigin()`